File: day3/two_number_sum.py
# Time Complexity: O(n)
# Space Complexity: O(n)
def twoNumberSum(array, targetSum):
    values = {}
    for item in array:
        targetValue = targetSum - item
        if targetValue in values:
            return [targetValue, item]
        else:
            values[item] = True
    return []

File: day3/two_number_sum.java
/* Time Complexity: O(n)
Space Complexity: O(n) */ 
import java.util.*;

class Program {
  public static int[] twoNumberSum(int[] array, int targetSum) {
    HashMap<Integer, Boolean> values = new HashMap<Integer, Boolean>();
    for(int i = 0; i< array.length; i++) {
      int targetValue = targetSum - array[i];
      if(values.get(targetValue) != null) {
        return new int[] {targetValue, array[i]};
      }
      else {
        values.put(array[i], true);
      }
    }
    return new int[] {};
  }
}
